.github/workflows/test.yaml

```diff
@@ -0,0 +1,40 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Test
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: mvn -B package --file pom.xml
```

Comment:
Take a look at the actions performed with this PR - the console output of this command. Note that `package` includes `test` so the next section is redundant.
I think the `--file pom.xml` is redundant. At least it is locally.
`-B` is a nice option (I didn't knew it existed). Add it to the next section too if you decide to keep it. To do that you'll have to skip the tests here (easy) and make the next section reuse the artifacts that this action delivers (not so easy); otherwise. I'll just drop `test1` if I were you.